src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -1515,13 +1515,28 @@ private function getErrorMessages(Exception $e)
                 'Only numbers and decimal points (".") are allowed in the amount fields, e.g. 10.50 or 1050.',
                 'Admin.Orderscustomers.Notification'
             ),
-            InvalidCartRuleDiscountValueException::class => sprintf(
-                '%s<ol><li>%s</li><li>%s</li><li>%s</li></ol>',
-                $this->trans('It looks like the value is invalid:', 'Admin.Orderscustomers.Notification'),
-                $this->trans('Percent or amount value must be greater than 0.', 'Admin.Orderscustomers.Notification'),
-                $this->trans('Percent value cannot exceed 100.', 'Admin.Orderscustomers.Notification'),
-                $this->trans('Discount value cannot exceed the total price of this order.', 'Admin.Orderscustomers.Notification')
-            ),
+            InvalidCartRuleDiscountValueException::class => [
+                InvalidCartRuleDiscountValueException::INVALID_MIN_PERCENT => $this->trans(
+                    'Percent value must be greater than 0.',
+                    'Admin.Orderscustomers.Notification'
+                ),
+                InvalidCartRuleDiscountValueException::INVALID_MAX_PERCENT => $this->trans(
+                    'Percent value cannot exceed 100.',
+                    'Admin.Orderscustomers.Notification'
+                ),
+                InvalidCartRuleDiscountValueException::INVALID_MIN_AMOUNT => $this->trans(
+                    'Amount value must be greater than 0.',
+                    'Admin.Orderscustomers.Notification'
+                ),
```

Comment:
Hi @jolelievre ! Is it necessary to have _"Percent value must be greater than 0."_ and _"Amount value must be greater than 0."_ both texts separated? 

If not, we can replace them by the string **_"Percent or amount value must be greater than 0."_** in _Admin.Orderscustomers.Notification_

Reply:
That was the initial text but it was used in a list because we had no way to know what was the cause of the error. We now have a much more precise exception which indicates precisely if the problem comes from a value or a percent.

I can use the same error message for both errors though if you prefer, I just thought more accurate error messages were better for the user

Reply:
@jolelievre indeed having accurate error messages are better for users! Let's keep both messages separated then. Thanks for explaining the context :)